openeogeotrellis/geopysparkdatacube.py

```diff
@@ -1222,15 +1222,27 @@ def resample_spatial(
             if newLayout == None:
                 return self
 
-            partitioner = max_level.srdd.rdd().partitioner()
-            b = max_level.layer_metadata.bounds
-            rows = b.maxKey.row - b.minKey.row + 1
-            spatialPartitions = (b.maxKey.col - b.minKey.col + 1) * (rows)
+            cellsize_before = self.get_cellsize()
+            if not isinstance(resolution, tuple):
+                resolution = (resolution, resolution)
+
+            estimated_size_in_pixels_tup = self.estimate_layer_size_in_pixels()
 
-            logging.info(f"Reprojecting datacube with partitioner {partitioner} to new layout {newLayout} and {projection}")
-            if(max_level.getNumPartitions() <= spatialPartitions and max_level.layer_type == gps.LayerType.SPACETIME and resolution < 0.5*self.get_cellsize_x()):
-                logging.info(f"Repartitioning datacube with {max_level.getNumPartitions()} partitions to {spatialPartitions*10} before resample_spatial.")
-                max_level = max_level.repartition(spatialPartitions*10)
+            resolution_factor = (
+                cellsize_before[0] / resolution[0],
+                cellsize_before[1] / resolution[1]
+            )
+
+            proposed_partition_count_tup = (
+                estimated_size_in_pixels_tup[0] * resolution_factor[0] / currentTileLayout.tileCols,
```

Comment:
I'm wondering if this formula is actually a more complex way to write something like:
number_of_spatial_tiles*resolution_factor^2
but anyway, perhaps doesn't matter as I would go for a memory based estimate in the final solution, that we perhaps do after merging this PR?

Reply:
It makes a slight difference when layer is not divisible by the tilesize. But is maybe not worth the complexity
In my SPI test it goes from `922` to `1128` when multiplying tile number directly.